proc-macros/src/new/lifetimes.rs

```diff
@@ -0,0 +1,118 @@
+use proc_macro2::Span;
+use syn::spanned::Spanned;
+use syn::{Lifetime, Path, Type};
+
+static LIFETIME: &str = "'a";
+
+pub fn replace_lifetimes(ty: &mut Type) -> bool {
+	let mut replaced = false;
+	traverse_type(ty, &mut replaced, replace_lifetime);
+	replaced
+}
+
+pub fn replace_lifetime(ty: &mut Type, replaced: &mut bool) {
+	use syn::{GenericArgument, PathArguments};
+
+	match ty {
+		Type::Path(p) => {
+			for segment in p.path.segments.iter_mut() {
+				if let PathArguments::AngleBracketed(ref mut ab) = segment.arguments {
+					let mut lifetimes = 0;
+
+					for arg in ab.args.iter_mut() {
+						match arg {
+							GenericArgument::Lifetime(lt) => {
+								*lt = Lifetime::new(LIFETIME, lt.span());
```

Comment:
Given that we repalce every lifetime we run into with "'a", I guess this will cause an error if there is more than one lifetime present.

Perhaps we could throw a nice compile error if we run into a second lifetime to make it clear what the issue in in that case?

Reply:
I was thinking about it, but I don't think that's going to come up in practice, you'd have to have two lifetimes defined on the method signature, and you could still produce a conflict by having them on separate types:

```rust
fn foo<'a, 'b>(&self, a: &'a str, b: &'b str) -> some_type
```

Those would be then put into the struct with lifetime `'a` and passed through, and I'm not even convinced there would be any error there necessarily.

Reply:
Correction: since we are using `async-trait` all lifetimes have to be either `'_`, or just one specific lifetime, so this isn't much of an issue.